### PR TITLE
[Feature] Update deploy test tools

### PR DESCRIPTION
**Pull request by grimoire**
Hi
This PR adds TensorRT deploy test tools and rename `ort_test.py` to `deploy_test.py`
Result below:

|   Model    |                     Config                      |  Dataset   | Metric | PyTorch | ONNXRuntime | TensorRT-fp32 | TensorRT-fp16 |
| :--------: | :---------------------------------------------: | :--------: | :----: | :-----: | :---------: | :-----------: | :-----------: |
|    FCN     |      fcn_r50-d8_512x1024_40k_cityscapes.py      | cityscapes |  mIoU  |  72.2   |    72.2     |     72.2      |     72.2      |
|   PSPNet   |    pspnet_r50-d8_512x1024_40k_cityscapes.py     | cityscapes |  mIoU  |  77.8   |    77.8     |     77.8      |     77.8      |
| deeplabv3  |   deeplabv3_r50-d8_512x1024_40k_cityscapes.py   | cityscapes |  mIoU  |  79.0   |    79.0     |     79.0      |     79.0      |
| deeplabv3+ | deeplabv3plus_r50-d8_512x1024_40k_cityscapes.py | cityscapes |  mIoU  |  79.6   |    79.5     |     79.5      |     79.5      |

### CODE REVIEW

File: tools/deploy_test.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+from mmcv.parallel import MMDataParallel
+from mmcv.runner import get_dist_info
+from mmcv.utils import DictAction
+
+from mmseg.apis import single_gpu_test
+from mmseg.datasets import build_dataloader, build_dataset
+from mmseg.models.segmentors.base import BaseSegmentor
+
+# import onnxruntime if needed
+try:
```
Comment (RunningLeon) [inline tools/deploy_test.py:19]: we could remove try exept and move import to line 232

File: tools/deploy_test.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+            return seg_pred
+
+        def aug_test(self, imgs, img_metas, **kwargs):
+            raise NotImplementedError('This method is not implemented.')
+
+except Exception:
+    pass
+
+# import onnxruntime if needed
+try:
```
Comment (RunningLeon) [inline tools/deploy_test.py:97]: we could remove try exept and move import to line 234

File: tools/deploy_test.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+except Exception:
+    pass
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description='mmseg backend test (and eval)')
+    parser.add_argument('config', help='test config file path')
+    parser.add_argument('model', help='Input model file')
+    parser.add_argument(
```
Comment (RunningLeon) [inline tools/deploy_test.py:155]: Can we determine the backend type by the file extension of input model file. For instance, '.onnx' for onnxruntime, and '.trt' for TensorRT?

Comment (grimoire) [reply tools/deploy_test.py:155]: Users might use `.engine`, `.model` or other extensions.

File: tools/deploy_test.py [hunk 4]
```diff
@@ -15,136 +15,129 @@
...
+            seg_pred = seg_pred.long().detach().cpu().numpy()
+        seg_pred = seg_pred[0]
+        seg_pred = list(seg_pred)
+        return seg_pred
+
+    def aug_test(self, imgs, img_metas, **kwargs):
+        raise NotImplementedError('This method is not implemented.')
+
 
 # import onnxruntime if needed
```
Comment (RunningLeon) [inline tools/deploy_test.py:94]: Unnecessary comments could be removed.

File: tools/deploy_test.py [hunk 5]
```diff
@@ -15,136 +15,129 @@
 from mmseg.datasets import build_dataloader, build_dataset
 from mmseg.models.segmentors.base import BaseSegmentor
 
+
 # import onnxruntime if needed
```
Comment (RunningLeon) [inline tools/deploy_test.py:19]: Unnecessary comments could be removed.

File: docs/useful_tools.md [hunk 6]
```diff
@@ -107,6 +110,7 @@ Description of all arguments
 
 - `config`: The path of a model config file.
 - `model`: The path of a ONNX model file.
```
Comment (RunningLeon) [inline docs/useful_tools.md:112]: Here could be updated since that both onnx file and TensorRT engine file could be the input.

File: docs/useful_tools.md [hunk 7]
```diff
@@ -88,12 +88,15 @@ We provide `tools/ort_test.py` to evaluate ONNX model with ONNXRuntime backend.
...
 
+- Install TensorRT python package(optional)
+
 #### Usage
 
 ```bash
-python tools/ort_test.py \
+python tools/deploy_test.py \
     ${CONFIG_FILE} \
     ${ONNX_FILE} \
```
Comment (RunningLeon) [inline docs/useful_tools.md:98]: `ONNX_FILE` could update to `MODEL_FILE`

File: docs/useful_tools.md [hunk 8]
```diff
@@ -88,12 +88,15 @@ We provide `tools/ort_test.py` to evaluate ONNX model with ONNXRuntime backend.
   pip install onnx onnxruntime-gpu
   ```
 
+- Install TensorRT python package(optional)
```
Comment (RunningLeon) [inline docs/useful_tools.md:91]: could add the doc link here for reference: https://mmcv.readthedocs.io/en/latest/tensorrt_plugin.html#how-to-build-tensorrt-plugins-in-mmcv

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by xvjiarui — APPROVED** [review]

**Comment by xvjiarui** [comment]
I suggest open a new documentation page for deployment, so users could easily find the exporting guide.
